app/src/main/kotlin/com/wire/android/media/audiomessage/RecordAudioMessagePlayer.kt

```diff
@@ -0,0 +1,222 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.android.media.audiomessage
+
+import android.content.Context
+import android.media.MediaPlayer
+import androidx.core.net.toUri
+import dagger.hilt.android.scopes.ViewModelScoped
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.merge
+import java.io.File
+import javax.inject.Inject
+
+@ViewModelScoped
+class RecordAudioMessagePlayer @Inject constructor(
```

Comment:
Can't we use `ConversationAudioMessagePlayer` instead of creating a similar one ?